tools/tplg_parser/tplg_parser.c

```diff
@@ -144,9 +144,15 @@ int tplg_load_buffer(int comp_id, int pipeline_id, int size,
 
 	tplg_read_array(array, file);
 
+	/* Ignore possible remaining vendor array data. TODO: Parse also
+	 * successive vendor arrays. In SOF topologies the buffer CORE_ID
+	 * token is passed there.
+	 */
+	fseek(file, size - array->size, SEEK_CUR);
+
```

Comment:
@singalsu why not fix it the right way with a while look through all arrays to find the buffer tokens just like we;ve done for some other components?

Reply:
that way u dont need to assum buffer tokens are first and you dont have to do the fseek manually either

Reply:
need to check return code, i.e. we could get eof or another error.

Reply:
Thanks for suggestion @ranj063 , in the new version I just followed the example of PCM widget and it was quite easy to do.

Reply:
@lgirdwood Yep, I could avoid this fseek() in the new version with the while loop for all arrays.

Reply:
@singalsu looks good now except for that one comment for Liam. And you need to squash the 2nd and 3rd patches?

Reply:
Yes I noticed the mistake of not amending updates into 2nd commit. It's fixed in this version. 

I missed the comment by Liam I'll fix ASAP.

Reply:
@lgirdwood @ranj063 The return code checks are now added to fourth commit.